myjsonrpc.pm

```diff
@@ -38,10 +38,10 @@ sub send_json ($to_fd, $cmd) {
     $json .= "\n";
 
     confess 'myjsonprc: called on undefined file descriptor' unless defined $to_fd;
-    my $wb = syswrite($to_fd, "$json");
-    if (!$wb || $wb != length($json)) {
-        die('myjsonrpc: remote end terminated connection, stopping') if !DEBUG_JSON && $! =~ qr/Broken pipe/;
-        confess "syswrite failed: $!";
+    my $wb = syswrite($to_fd, "$json"); # wb = write buffer
+    if (!$wb || $wb != length($json)) { # is true when wb is empty or when the lenght of wb is not the entire json lenght
```

Comment:
Rather than commenting stuff we should aim for making the code clearer, for example by using descriptive variable names.
In this case it seemed to cause a misunderstanding. `$wb` is not the buffer, but the return value of `syswrite`, which is `Returns the number of bytes actually written`.
With that I hope we can go without a comment. `!$number` is basically short for `$number == 0` in perl:
```suggestion
    my $written_bytes = syswrite($to_fd, "$json");
    if (!$written_bytes || $written_bytes != length($json)) {
```

Reply:
As a draft this is not a ready pr, I'll remove the comment while refactoring :+1: 

I thought that `!$wb` was used to store the output, thanks for the hint :pray: I'll rename the variable as you suggest :+1: